zstacklib/zstacklib/utils/gpu.py

```
@@ -14,6 +17,74 @@ class VmGpuStatus(Enum):
+    if not domain or not domain.isActive():
+        logger.info("no need to shutdown nvidia-persistenced for vm %s, it is not running" % vm_uuid)
+        return 0, None
+    else:
+        logger.info("start to shutdown nvidia-persistenced for vm %s" % vm_uuid)
+        return shut_persistenced_by_guesttool(domain)
+
+
+def nvidia_pre_detach_from_host_hook():
+    """NVIDIA specific pre-detach-from-host hook"""
+    logger.info("start to shutdown nvidia-persistenced on host")
+    r, o, _ = bash_roe(get_shut_nvidia_persistence_cmd())
```

Ruff (0.14.4)

`bash_roe` may be undefined, or defined from star imports

(F405)